{"id": 5243, "title": "Ticket #5243: Avoid DAGMan restarts on dprintf failure, etc.", "description": "<blockquote>\nWe ran into this problem on CHTC:\n\n<p>What was the problem you ask?  User \"smallegan\" had three dagman jobs that were starting, exiting, and being restarted over a million times because the user filled up his disk quota at 12:30pm.  DAGMan could not write to its logs and so it immediately exited (with exitcode 44, which is what all HTCondor daemons do when they cannot write to their logs), at which point the schedd immediately restarted it as fast as it could.   I put user \"smallegan\"'s jobs on hold and the duty cycle immediately dropped and all was happy again.\n\n</p><p>Just discussed this with ToddT -- he suggests that DAGMan should be removed from the queue on exit with anything other than SIGTERM, SIGKILL, or SIGQUIT.  My initial thought is that that's probably fine, but I want to investigate a bit before saying yes for sure.\n\n</p><p>(Part of the idea is that we don't just want to say, \"DAGMan should be removed from the queue if it exits with exit code 44, because there might be similar issues in the future that would not cause that specific exit code.)</p></blockquote>", "remarks": "<blockquote>\n<em>2015-Aug-28 11:48:15 by wenger:</em> <br/>\n\nOkay, exit status 3 is what we get on condor_hold of DAGMan, so we definitely want to stay in the queue in that case...\n\n<p></p><hr/>\n<em>2015-Aug-28 12:27:33 by wenger:</em> <br/>\n\nHmm -- the schedd seems to ignore <span class=\"wiki\"><a class=\"missing\" href=\"wiki?p=OnExitRemove\" title=\"On Exit Remove\">OnExitRemove</a></span> when condor_hold is done on a scheduler universe job.  I set <span class=\"wiki\"><a class=\"missing\" href=\"wiki?p=OnExitRemove\" title=\"On Exit Remove\">OnExitRemove</a></span> to true, but condor_dagman still stays in the queue after being condor_held...\n\n<p></p><hr/>\n<em>2015-Aug-28 14:56:35 by wenger:</em> <br/>\n\nCreated V8_5-gittrac_5243-branch for this.\n\n<p></p><hr/>\n<em>2015-Aug-28 15:38:39 by wenger:</em> <br/>\n\nOkay, I've committed a fix on the branch that removes DAGMan from the queue on an exit code of 44.  That will fix the immediate problem with re-starts on dprintf failures, and I'm 99% certain that it doesn't break anything.\n\n<p>My thought is that if we want something out in the next few days we should go with this, and then revisit a more comprehensive fix when we have a little more time to think about it.</p></blockquote>", "derived_tickets": "", "attachments": "", "check_ins": "<table border=\"0\" cellpadding=\"0\" cellspacing=\"0\">\n<tbody><tr><td align=\"right\" valign=\"top\" width=\"160\">2015-Aug-28 15:36</td>\n<td align=\"center\" valign=\"top\" width=\"30\">\n<span class=\"icon dot\">\u00a0</span></td>\n<td align=\"left\" valign=\"top\"> \nCheck-in <span class=\"chng\"><a href=\"chngview?cn=45756\">[45756]</a></span>: Gittrac <span class=\"ticket\"><a class=\"stalled\" href=\"/tickets?ticket=5243\" onclick=\"get_ticket_and_populate_wrapper('5243'); return false;\" title=\"Avoid DAGMan restarts on dprintf failure, etc.\">#5243</a></span>: Tentative fix -- we remove DAGMan from the queue on exit code 44 (dprintf failure).  (By Kent Wenger )</td></tr>\n</tbody></table>", "type": "todo", "last_change": "2018-Oct-19 13:56", "status": "stalled", "created": "2015-Aug-28 11:01", "fixed_version": "2015-Aug-28 11:01", "broken_version": "v080308", "priority": "2", "subsystem": "Dag", "assigned_to": "coatsworth", "derived_from": "", "creator": "wenger", "rust": "", "customer_group": "chtc", "visibility": "public", "notify": "wenger@cs.wisc.edu, zmiller@cs.wisc.edu, tannenba@cs.wisc.edu, coatsworth@cs.wisc.edu", "due_date": ""}